### ISSUE TITLE
too many try catch break unit test reliability

### ISSUE THREAD

**Issue by cvandeplas** [EC-DIGIT-CSIRC/sysdiagnose#121]
the try catch with logger.exception break unit test reliability. 
We need to do this better.

**Comment by dario-br** [comment]
@cvandeplas Isn't this address by the PR #235 ?

**Comment by cvandeplas** [comment]
yes indeed, I am still a bit puzzled and have mixed feelings with the big try catch though.
The check for consistency in the report compensates, but it made troubleshooting issues more complex. 
We may want to refactor that in the near future.
